x11-terms/blackbox-terminal/blackbox-terminal-0.14.0.ebuild

```
@@ -0,0 +1,70 @@
+# Copyright 2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v3 or later
+
+EAPI=8
+
+inherit flag-o-matic meson gnome2-utils xdg vala
+
+DESCRIPTION="A beautiful GTK 4 terminal"
+HOMEPAGE="https://gitlab.gnome.org/raggesilver/blackbox"
+SRC_URI="https://gitlab.gnome.org/raggesilver/blackbox/-/archive/v${PV}/blackbox-v${PV}.tar.bz2
+	-> ${P}.tar.bz2"
+
+S="${WORKDIR}/blackbox-v${PV}"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	>=dev-libs/glib-2.0:2
+	>=dev-libs/json-glib-1.0
+	>=dev-libs/libpqmarble-2.0.0
+	>=gui-libs/gtk-4.0:4
+	>=gui-libs/libadwaita-1.0:1
+	>=dev-libs/libgee-0.20:0.8
+	>=gui-libs/vte-0.70:2.91-gtk4
+	x11-libs/pango
+"
+RDEPEND="${DEPEND}"
+
+BDEPEND="
+	>=dev-lang/vala-0.56
+	dev-libs/gobject-introspection
+	virtual/pkgconfig
+	sys-devel/gettext
+"
+
+# GCC 14 compatibility: Replace return*_if_fail macros
+# https://gitlab.gnome.org/raggesilver/blackbox/-/commit/e50457a52aafcdfe0e230a3e6ec97eddc66eb989
+PATCHES=(
+	"${FILESDIR}/${PN}-0.14.0-gcc-14.patch"
+)
+
+src_prepare() {
+	default
+	vala_setup
+}
+
+src_configure() {
+	# Workaround for Vala-generated C code pointer type warnings
+	# See: https://gitlab.gnome.org/GNOME/vala/-/issues/1408
+	#      https://bugs.gentoo.org/892708
+	append-cflags -Wno-incompatible-pointer-types
+
+	meson_src_configure -Dblackbox_is_flatpak=false
+}
```

```suggestion
src_configure() {
	meson_src_configure -Dblackbox_is_flatpak=false
}
```



vala_setup 中已经添加过 -Wno-incompatible-pointer-types ，所以不需要再 append 一次